File: INTRODUCCION A POO/EJERCICIO-3/Coche.py
class Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0

    def acelerar(self):
        self.velocidad += 10

    def frenar(self):
        self.velocidad = max(0, self.velocidad - 5)

    def __str__(self):
        return f"{self.marca} {self.modelo} - Velocidad: {self.velocidad} km/h"


coche1 = Coche("Toyota", "Corolla")
coche2 = Coche("Ford", "Mustang")

coche1.acelerar()
coche1.acelerar()

coche1.frenar()
coche2.frenar()

print(f"Coche 1: {coche1.marca} {coche1.modelo}, Velocidad: {coche1.velocidad} km/h")
print(f"Coche 2: {coche2.marca} {coche2.modelo}, Velocidad: {coche2.velocidad} km/h")

File: INTRODUCCION A POO/EJERCICIO-3/Coche/src/ejercicio2/Coche.java
package ejercicio2;

public class Coche {
    private String marca;
    private String modelo;
    private int velocidad;
    
    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0;
    }
    
    public void acelerar() {
        this.velocidad += 10;
    }
    
    public void frenar() {
        this.velocidad = Math.max(0, this.velocidad - 5);
    }
    
    @Override
    public String toString() {
        return marca + " " + modelo + " - Velocidad: " + velocidad + " km/h";
    }
    
    public static void main(String[] args) {
        Coche coche1 = new Coche("Toyota", "Corolla");
        Coche coche2 = new Coche("Ford", "Mustang");
        
        coche1.acelerar();
        coche2.acelerar();
        
        coche1.frenar();
        coche2.frenar();
        
        System.out.println("Coche 1: " + coche1.marca + " " + coche1.modelo + 
                          ", Velocidad: " + coche1.velocidad + " km/h");
        System.out.println("Coche 2: " + coche2.marca + " " + coche2.modelo + 
                          ", Velocidad: " + coche2.velocidad + " km/h");
    }
}